lib/sequel/plugins/enum_values.rb

```diff
@@ -68,20 +70,30 @@ def enum_values(field)
 
 				private
 
-				def transform_predicate_methods_to_enum_fields(predicate_methods)
-					case predicate_methods
-					when TrueClass
-						all_enum_fields.keys
-					when FalseClass
-						[]
-					else
-						Array(predicate_methods)
+				def process_enum_values_options
+					@enum_values_caching =
+						@enum_values_options.fetch(:caching, @enum_values_caching)
+
+					process_enum_values_option(:filter_methods) do |field, value|
+						dataset.methods.include?(value.to_sym) &&
+							warn("WARNING: Redefining method #{value}")
+
+						dataset_module { where value, field => value }
+					end
+
+					process_enum_values_option(:predicate_methods) do |field, value|
+						define_method("#{value}?", -> { public_send(field) == value })
```

Comment:
```suggestion
						define_method("#{value}?") { public_send(field) == value }
```

But is here a really need to shorten this non-related code?